### DISCUSSION TITLE
Generate requests-mock from open api spec

### DISCUSSION THREAD

**Discussion by noorul** [schemathesis/schemathesis#1685] [Q&A] [answered]
I know that this tool is used to test against functional endpoints. Is it possible to generate [requests-mock](https://pypi.org/project/requests-mock/) from open an open API spec?

**Comment by Stranger6667** [comment]
Hi! Could you please, elaborate a bit? What is your use case?

**Comment by noorul** [comment]
We have a service which has multiple rest endpoints. They also publish spec. We have a handwritten python rest client for those APIs. We would like to mock rest endpoints (requests-mock) in our test cases according to the spec to validate whether the client is breaking any specification.

**Comment by Stranger6667** [comment, marked as answer]
Understood. I had this idea for some time and even wanted to implement it as a separate project / submodule for Schemathesis. But I think you could use `hypothesis-jsonschema` instead and generate values based on your schema, however, it does not do any mutations as Schemathesis does to generate negative test cases. You could hook into its internal APIs, but I won't give any guarantees that it won't change

**Comment by noorul** [comment]
Thank you for pointing to hypothesis-jsonschema. I had a cursory look and it needs more work to make it work for my use case compared to this project. If someone has to think about an extension for this project where should they start reading the code?

**Reply by Stranger6667** [reply]
You’re welcome! I’d suggest taking a look at the functions used in this dict - https://github.com/schemathesis/schemathesis/blob/511999e5079bd7a08f90ad6dab25995553316638/src/schemathesis/specs/openapi/_hypothesis.py#L382
